src/transforms/reduce/transform.rs

```diff
@@ -760,4 +765,139 @@ merge_strategies.bar = "concat"
+
+            tx.send(e_1.into()).await.unwrap();
+
+            let e_2 = LogEvent::from(Value::from(btreemap! {
+                "id" => 777,
+                "message" => btreemap! {
+                        "a" => btreemap! {
+                            "b" => [3,4]
+                        }
+                },
+                 "arr" => vec![btreemap! { "a" => 2 }, btreemap! { "b" => 2 }],
+                "test_end" => "done",
+            }));
+            tx.send(e_2.into()).await.unwrap();
+
+            let mut output = out.recv().await.unwrap().into_log();
+
+            // Remove timestamp fields which were automatically added.
+            output.remove_timestamp();
+            output.remove("timestamp_end");
+
+            assert_eq!(
+                *output.value(),
+                btreemap! {
+                    "message" => btreemap! {
+                        "a" => btreemap! {
+                            "b" => vec![[1, 2], [3,4]],
+                        }
+                },
+                    "id" => 777,
+                     "arr" => vec![btreemap! { "a" => 1 }, btreemap! { "b" => 1 }],
+                    "test_end" => "done",
+                }
+                .into()
+            );
+
+            drop(tx);
+            topology.stop().await;
+            assert_eq!(out.recv().await, None);
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn strategy_path_with_indexed_field() {
+        let reduce_config = toml::from_str::<ReduceConfig>(indoc!(
+            r#"
+        group_by = [ "id" ]
+
+        merge_strategies.id = "discard"
+        merge_strategies."arr[0]" = "concat"
+
+        [ends_when]
+          type = "vrl"
+          source = "exists(.test_end)"
+        "#,
+        ))
+        .unwrap();
+
+        assert_transform_compliance(async move {
+            let (tx, rx) = mpsc::channel(1);
+
+            let (topology, mut out) = create_topology(ReceiverStream::new(rx), reduce_config).await;
+
+            let e_1 = LogEvent::from(Value::from(btreemap! {
+                "id" => 777,
+                "arr" => vec![ vec![1] ]
+            }));
+            let mut metadata_1 = e_1.metadata().clone();
+            metadata_1.set_upstream_id(Arc::new(OutputId::from("reduce")));
+
+            tx.send(e_1.into()).await.unwrap();
+
+            let e_2 = LogEvent::from(Value::from(btreemap! {
+                "id" => 777,
+                "arr" => vec![ vec![2] ],
+                "test_end" => "done",
+            }));
+            tx.send(e_2.into()).await.unwrap();
+
+            let mut output = out.recv().await.unwrap().into_log();
+
+            // Remove timestamp fields which were automatically added.
+            output.remove_timestamp();
+            output.remove("timestamp_end");
+
+            assert_eq!(
+                *output.value(),
+                btreemap! {
+                    "id" => 777,
+                     "arr" => vec![ vec![ 1 ]],
```

Comment:
This is not an expected behaviour for `concat` on arrays - compare with the `arrays` test. The result here should be `"arr" => vec![vec![ 1, 2 ]]`.

Reply:
This test is now removed (https://github.com/vectordotdev/vector/pull/20800/commits/a349551cd01a48bfb17d3806ea25c23c4ef84a96). As explained in the PR description, I added this to demonstrate the existing behavior i.e. we do not traverse individual array elements. [See PR](https://github.com/vectordotdev/vector/pull/20813) containing this test in isolation.

As we agreed offline, it's cleaner to reject configs (done in https://github.com/vectordotdev/vector/pull/20800/commits/a349551cd01a48bfb17d3806ea25c23c4ef84a96) with indexed paths to avoid confusion. This behavior can be revisited in the future 👍